### DISCUSSION TITLE
Phantom Wallet connector

### DISCUSSION THREAD

**Discussion by thenoahhein** [wevm/wagmi#1346] [Ideas]
Hey Wagmi team,

I'm a developer advocate for Phantom wallet, and we're moving over to ETH/Polygon. We have a closed beta currently, and a few dapps we're working with are trying to add Phantom to their wallet modals. I have a [PR open](https://github.com/rainbow-me/rainbowkit/pull/875) for rainbowkit for example.

I wanted to see if y'all would be open to the idea of me writing a PR to add Phantom as one of the wallet connectors? This will help the downstream packages such as rainbowkit that rely on wagmi's connector packages.

I'm happy to do all of the work related to this, I just didn't want to do the work on it before seeing if this was something the team would be open to.

**Comment by tmm** [comment]
Thanks for starting this thread! Let's continue this conversation when Phantom wallet for EVM is out of closed beta and generally available.

**Reply by bfriel** [reply]
Hey @tmm! Phantom has now opened up its multi-chain waitlist to the public. We're really excited about the [reception to our announcement](https://twitter.com/phantom/status/1597597365774749697) and have received a ton of inbounds from EVM dapps looking to integrate with us.

A key component of our rollout is ensuring that we are properly integrated with EVM dApps. wagmi is a critical dependency for this current phase of our rollout. As @nheingit mentioned, we are more than happy to do the heavy lifting here. Are you open to us submitting a PR and integrating with wagmi?

**Reply by tmm** [reply]
Thanks for the update. Assuming Phantom is an injected wallet from the linked RainbowKit PR above, downstream libraries, like RainbowKit, can implement support without needing a Connector in wagmi. For example:

```ts
const phantomConnector = new InjectedConnector({
  options: {
    getProvider() {
      const getPhantom = (ethereum?: Ethereum) =>
        ethereum?.isPhantom ? ethereum : undefined
      if (typeof window === 'undefined') return
      if (window.ethereum?.providers)
        return window.ethereum.providers.find(getPhantom)
      return getPhantom(window.ethereum)
    },
    name: 'Phantom',
  },
})
```

**Reply by thenoahhein** [reply]
Currently Phantom is an injected wallet, and what you've done above is what I did for the Rainbowkit PR. However we're working with the WalletConnect team on phantom support, and discussing with them what that will look like.

And while I brought up rainbowkit as a concrete usecase that would be made easier by a wagmi integration, it is __not__ blocking the rainbowkit PR, or why I opened this discussion. There are several dapps such as [Foundation](https://foundation.app/) that have a custom wallet UI, but is completely powered by wagmi hooks behind the scenes. 

We're hoping with a Phantom integration that those dapps will be able to integrate Phantom with minimal overhead.

**Reply by tmm** [reply]
Got it. Thanks for the additional context!

Wallets do not map directly to connectors and are a higher-level concept. Often if you want to add cross-platform support for a wallet, you need to compose multiple connectors together. For example, developers adding support for Phantom could use the `InjectedConnector` on desktop and `WalletConnectConnector` on mobile. RainbowKit and other libraries have built out abstractions for this. At some point, wagmi might include this type of abstraction, but it is not on the roadmap right now.

Happy to set up some time to chat live dev@wagmi.sh.

**Reply by thenoahhein** [reply]
Totally, I'll shoot you an email to coordinate a good time to talk.

Thanks for the background, I'll take it back to the team to discuss before our meet.

And really appreciate the quick response 🙏

**Comment by codingwithmanny** [comment]
Would love to see this too to diversify wallet options.

**Comment by Nilbristi** [comment]
@nheingit **Not working in "@wagmi/core": "^2.0.2"**
This connector is not working with latest wagmi/core version.
Can you please update it?

I am following this link https://github.com/phantom/phantom-wagmi-connector

**Reply by 424363283** [reply]
> @nheingit**在“@wagmi/core”中禁用：“^2.0.2”** 此连接器无法与最新的wagmi/core版本配合使用。 您可以更新吗？
> 
> 我正在关注此链接https://github.com/phantom/phantom-wagmi-connector
Hello, have you solved the problem of using wagmi to access phantom? I would like to know how to solve it, thank you
